nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/StreamBulkScroller.java

```diff
@@ -120,10 +121,11 @@ public void processRecord(ComputationContext context, String inputStreamName, Re
         }
 
         protected void processRecord(ComputationContext context, Record record) {
-            KeyValueStore kvStore = getKeyValueStore();
+            KeyValueStore kvStore = Framework.getService(KeyValueService.class).getKeyValueStore(BULK_KV_STORE_NAME);
             try {
-                BulkCommand command = getBulkCommandJson(record.data);
-                if (!kvStore.compareAndSet(record.key + STATE, SCHEDULED.toString(), BUILDING.toString())) {
+                String bulkActionId = record.getKey();
+                BulkCommand command = BulkCommandHelper.getBulkCommandJson(record.getData());
+                if (!kvStore.compareAndSet(bulkActionId + STATE, SCHEDULED.toString(), BUILDING.toString())) {
```

Comment:
This should not happen, it will mean that 2 consumers of the same group read the same partition. But this check can be useful for handling cancellation, anyway if you skip the record you need to call context.askForCompletion

Reply:
Will keep the compareAndSet and add `context.askForCompletion()`.